app/models/repo_solr_client.rb

```diff
@@ -38,4 +54,25 @@ def get_solr_doc_for(uuid)
     mms_client = MMSClient.new(mms_url: Rails.application.secrets.mms_url)
     mms_client.repoapi_solr_doc_for(uuid)
   end
+  
+  def update_index_and_delete_empty_parents(new_document)
+    return unless @repo_solr_client
+
+    old_document = get_doc(new_document[:uuid])["docs"].first
+
+    if old_document
+      # Delete old non-matching parentUUIDs
+      old_uuids_to_delete = old_document['parentUUID'] - new_document[:parentUUID]
```

Comment:
I guess you checked this but just wanted to make sure these are always going to be arrays